controller/src/main/java/io/pravega/controller/server/bucket/BucketManager.java

```diff
@@ -137,40 +183,56 @@ public void failed(State from, Throwable failure) {
     @Override
     protected void doStop() {
         log.info("{}: Stop request received for bucket manager", serviceType);
-        Collection<BucketService> tmp;
+        Set<Integer> tmp;
         synchronized (lock) { 
-            tmp = buckets.values();
+            tmp = buckets.keySet();
         }
-        
-        Futures.allOf(tmp.stream().map(bucketService -> {
-            CompletableFuture<Void> bucketFuture = new CompletableFuture<>();
-            bucketService.addListener(new Listener() {
-                @Override
-                public void terminated(State from) {
-                    super.terminated(from);
-                    bucketFuture.complete(null);
-                }
+        stopBucketServices(tmp);
+        stopBucketOwnershipListener();
+        stopLeader();
+    }
 
-                @Override
-                public void failed(State from, Throwable failure) {
-                    super.failed(from, failure);
-                    bucketFuture.completeExceptionally(failure);
-                }
-            }, executor);
-            bucketService.stopAsync();
-
-            return bucketFuture;
-        }).collect(Collectors.toList()))
-                .whenComplete((r, e) -> {
-                    stopBucketOwnershipListener();
-                    if (e != null) {
-                        log.error("{}: bucket service shutdown failed with exception", serviceType, e);
-                        notifyFailed(e);
-                    } else {
-                        log.info("{}: bucket service stopped", serviceType);
-                        notifyStopped();
-                    }
-                });
+    public void stopBucketServices(Set<Integer> bucketIds) {
```

Comment:
method pretty long for stop. can have some doc.

Reply:
Added java doc for this method.